packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -197,11 +370,98 @@ export class Table extends Resource {
       actions: ['dynamodb:ListStreams'],
       resourceArns: ['*'],
     });
- }
+  }
+
+  /**
+   * Creates a Table construct that represents an external table via table name.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param tableName The table's name.
+   */
+  public static fromTableName(scope: Construct, id: string, tableName: string): ITable {
+    return Table.fromTableAttributes(scope, id, { tableName });
+  }
+
+  /**
+   * Creates a Table construct that represents an external table via table arn.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param tableArn The table's ARN.
+   */
+  public static fromTableArn(scope: Construct, id: string, tableArn: string): ITable {
+    return Table.fromTableAttributes(scope, id, { tableArn });
+  }
+
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs A `TableAttributes` object.
+   */
+  public static fromTableAttributes(scope: Construct, id: string, attrs: TableAttributes): ITable {
+
+    class Import extends TableBase {
+      public readonly tableName: string;
+      public readonly tableArn: string;
+
+      constructor(_scope: Construct, _id: string, props: TableAttributes) {
+        super(_scope, _id);
+        this.tableArn = props.tableArn!;
+        this.tableName = props.tableName!;
+      }
+
+      protected get hasIndex(): boolean {
+        return false;
+      }
+    }
+
+    const stack = Stack.of(scope);
+    let tableName = '';
+    let tableArn = '';
+    // if we have an explicit table name, use it.
+    if (attrs.tableName) {
+      tableName = attrs.tableName;
+      if (!attrs.tableArn) {
+        tableArn = stack.formatArn({
```

Comment:
This should be:

```ts
tableArn = stack.formatArn({
  service: 'dynamodb',
  resource: 'table',
  resourceName: attrs.tableName,
});
```

Reply:
Per the doc https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazondynamodb.html#amazondynamodb-resources-for-iam-policies, are `region` and `account` are mandatory for a valid DynamoDB ARN?

Reply:
Right. It's just that if you don't pass anything for `account` and `region`, defaults will be used (see here: https://github.com/aws/aws-cdk/blob/2df88f3481998db98d3bbd950998f1c0bf5e3b9c/packages/%40aws-cdk/core/lib/arn.ts#L81-L82 ).

So no reason to pass these explicitly.